torchmetrics/utilities/exceptions.py

```diff
@@ -0,0 +1,25 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+class BaseUserError(Exception):
+    """
+    Base user error class to indicate wrong usage.
+    """
+
+
```

Comment:
n00b q: why do we need a base user error? instead of various user errors coming from Exception?

Reply:
Removed ! Suggestion from Justus.